content/tutorial/tutorial.md

```diff
@@ -750,41 +757,41 @@ We can now change the Board's `handleClick` function to return early by ignoring
   }
 ```
 
-**[View the full code at this point](https://codepen.io/gaearon/pen/LyyXgK?editors=0010)**
+**[अभी तक का पूरा कोड देखें](https://codepen.io/gaearon/pen/LyyXgK?editors=0010)**
 
-Congratulations! You now have a working tic-tac-toe game. And you've just learned the basics of React too. So *you're* probably the real winner here.
+बधाई हो! अब आपके पास काम करने वाला का टिक-टैक-टो खेल है। और आपने अभी-अभी React के बेसिक्स भी सीखे हैं। तो *आप* यहां के असली विजेता हैं।
 
-## Adding Time Travel {#adding-time-travel}
+## टाइम ट्रेवल गेम में डालना {#adding-time-travel}
 
-As a final exercise, let's make it possible to "go back in time" to the previous moves in the game.
+अंतिम अभ्यास के रूप में, चलिए "समय में पीछे जाना" खेल की पिछली चालों तक संभव बनाते है।
 
-### Storing a History of Moves {#storing-a-history-of-moves}
+### मूव्स की हिस्ट्री स्टोर करना {#storing-a-history-of-moves}
 
-If we mutated the `squares` array, implementing time travel would be very difficult.
+यदि हम `Square` array को बदल देते हैं, तो टाइम ट्रेवल को लागू करना बहुत मुश्किल होगा।
 
-However, we used `slice()` to create a new copy of the `squares` array after every move, and [treated it as immutable](#why-immutability-is-important). This will allow us to store every past version of the `squares` array, and navigate between the turns that have already happened.
+हालाँकि, हमने हर चाल के बाद `squares` array की एक नई प्रति बनाने के लिए `slice()` का उपयोग किया, और [इसे अपरिवर्तनीय माना](#why-immutability-is-important)। यह हमें `squares` array के हर पिछले वर्शन को स्टोर करके रखने में मदद करेगा, और पिछली चालों के बीच नेविगेट करने देगा जो पहले ही हो चुके हैं।
 
-We'll store the past `squares` arrays in another array called `history`. The `history` array represents all board states, from the first to the last move, and has a shape like this:
+हम `history` नामक एक और array में पिछले `squares` arrays को संग्रहीत करेंगे। `history` array पहली से अंतिम चाल तक सभी Board स्टेट्स का प्रतिनिधित्व करता है, और इस तरह से इसके पास एक आकार है:
```

Comment:
और इस तरह से इसके पास एक आकार है: does not seem right?

Reply:
> और इस तरह से इसके पास एक

I think we can drop this line.  Just till this looks fine I guess `Board स्टेट्स का प्रतिनिधित्व करता है`
any thoughts

Reply:
could be reworded to aur kuch aisa dikhta hai?